pkg/cstor/cvr_operation.go

```diff
@@ -67,12 +75,61 @@ func (p *Plugin) waitForAllCVR(vol *Volume) error {
 		}
 		time.Sleep(CVRCheckInterval)
 	}
+	return errors.Errorf("CVR for volume{%s} are not ready!", vol.volname)
+}
+
+// waitForCSIBasedCVRs will ensure that all CVRs related to
+// the given CSI volume is created.
+func (p *Plugin) waitForCSIBasedCVRs(vol *Volume, replicaCount int) error {
+	for cnt := 0; cnt < CVRWaitCount; cnt++ {
+		cvrList, err := p.OpenEBSAPIsClient.
+			CstorV1().
+			CStorVolumeReplicas(p.namespace).
+			List(metav1.ListOptions{
+				LabelSelector: "openebs.io/persistent-volume=" + vol.volname,
+			})
+		if err != nil {
+			return errors.Errorf("Failed to fetch CVR for volume=%s %s", vol.volname, err)
+		}
+
+		if len(cvrList.Items) != replicaCount {
+			time.Sleep(CVRCheckInterval)
+			continue
+		}
 
+		cvrCount := 0
+		for _, cvr := range cvrList.Items {
+			if cvr.Status.Phase == cstorv1.CVRStatusOnline ||
```

Comment:
this is more of a qn to check only.. what will be the state of CVR if targetIP is not set? cc: @mittachaitu

Reply:
It will be in error state @vishnuitta

Reply:
State will be `Error`

Reply:
Basically, if the replica is not connected to the target we will get empty stats
```sh
zfs stats <volume_dataset>
{"stats":[{"pool":{}}]}
```
Above will output will  unmarshal in the control plane and it will be 
```sh
Empty volume status for volume stats: '&{Stats:[{Name: Status: RebuildStatus: IsIOAckSenderCreated:0 IsIOReceiverCreated:0 RunningIONum:0 CheckpointedIONum:0 DegradedCheckpointedIONum:0 CheckpointedTime:0 RebuildBytes:0 RebuildCnt:0 RebuildDoneCnt:0 RebuildFailedCnt:0 Quorum:0}]}'
```
We have a [ZFSStatusTOCVRStatusMapper](https://github.com/openebs/cstor-operators/blob/d6c74a0a6a282143f593026c28d05be6fe5f6a96/pkg/volumereplica/volumereplica.go#L553) to return status as Error.